src/api/handlers.go

```diff
@@ -52,6 +53,33 @@ func ReadRequestBody(w http.ResponseWriter, r *http.Request, resp *interface{})
 	return
 }
 
+func ReadSessionStore(w http.ResponseWriter, r *http.Request, resp *interface{}, name string) (session *sessions.Session, err error) {
```

Comment:
Is there a better type for the `resp *interface{}`?

Reply:
That resp is changing by itself is a side-effect. In theory we can remove the param and add it as return value, sould i do that (i would do that everywhere)?

Reply:
Yeah changing that `resp` to a return value might make it more clear and avoid side effects. Does the type of `resp` change? I'm not sure why it has to be an interface here.

Reply:
no, type `interface{}` is a wildcard. resp contains the return value back to the client. In most cases it is a string or an object, that will be jsonalized before send.

Reply:
I usually try and stay away from generic interfaces because you lose all type safety. But if it works then I guess it's alright :shrug: